debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java

```diff
@@ -180,19 +275,36 @@ public OracleVersion getOracleVersion() {
         return oracleVersion;
     }
 
+    public String getSchemaName() {
+        return schemaName;
+    }
+
+    public Tables.ColumnNameFilter getColumnFilter() {
+        return columnFilter;
+    }
+
     @Override
     protected HistoryRecordComparator getHistoryRecordComparator() {
         return new HistoryRecordComparator() {
             @Override
             protected boolean isPositionAtOrBefore(Document recorded, Document desired) {
-                final LcrPosition recordedPosition = LcrPosition.valueOf(recorded.getString(SourceInfo.LCR_POSITION_KEY));
-                final LcrPosition desiredPosition = LcrPosition.valueOf(desired.getString(SourceInfo.LCR_POSITION_KEY));
-                final Long recordedScn = recordedPosition != null ? recordedPosition.getScn() : recorded.getLong(SourceInfo.SCN_KEY);
-                final Long desiredScn = desiredPosition != null ? desiredPosition.getScn() : desired.getLong(SourceInfo.SCN_KEY);
-
-                return (recordedPosition != null && desiredPosition != null)
-                        ? recordedPosition.compareTo(desiredPosition) < 1
-                        : recordedScn.compareTo(desiredScn) < 1;
+                Long recordedScn;
```

Comment:
Could be as a follow-up introduces a Scn domain type?

Reply:
I've added this as [DBZ-2518](https://issues.redhat.com/browse/DBZ-2518)